src/iuse.cpp

```diff
@@ -2117,30 +2163,212 @@ int iuse::sew_advanced(player *p, item *it, bool, const tripoint& )
     rn -= mod_count * 10;                              // Other mods
 
     if( rn <= 8 ) {
-        p->add_msg_if_player(m_bad, _("You damage your %s trying to modify it!"),
-                             mod->tname().c_str());
-        if( mod->inc_damage() ) {
-            p->add_msg_if_player(m_bad, _("You destroy it!"));
-            p->i_rem_keep_contents( pos );
-        }
-        return thread_needed / 2;
+
+		filament_left = filament_needed / 2;
+
+		// Go through all filament items and spend available until finished
+		for (auto &filament : mod_filament) {
+
+			auto inv = p->crafting_inventory();
+			auto itm = inv.item_by_type(filament);
+
+			bool filament_count_by_charges = itm.count_by_charges();
+
+			filament_have = (filament_count_by_charges) ? crafting_inv.charges_of(filament) : crafting_inv.amount_of(filament);
+
+			if (filament_have == filament_left)
+			{
+				filament_used = filament_have;
+			}
+			else if (filament_have > filament_left)
+			{
+				filament_used = filament_left;
+			}
+			else if (filament_have < filament_left)
+			{
+				filament_used = filament_have;
+			};
```

Comment:
All this block looks equivalent to `filament_used = std::min( filament_have, filament_left )`.